src/SriServiceProvider.php

```diff
@@ -29,39 +30,89 @@ public function boot()
         ]);
 
         Blade::directive('mixSri', function (string $path, bool $crossOrigin = false) {
+            $path = $this->clean_path($path);
+
             if (Str::startsWith($path, ['http', 'https', '//'])) {
                 $href = $path;
             } else {
                 $href = mix($path);
             }
 
-            $integrity = SriFacade::html($path, $crossOrigin);
-
-            if (Str::endsWith($path, 'css')) {
-                return "<link href='{$href}' rel='stylesheet' {$integrity}>";
-            } elseif (Str::endsWith($path, 'js')) {
-                return "<script src='{$href}' {$integrity}></script>";
-            } else {
-                throw new \Exception('Invalid file');
-            }
+            return $this->generate($path, $href, $crossOrigin);
         });
 
         Blade::directive('assetSri', function (string $path, bool $crossOrigin = false) {
+            $path = $this->clean_path($path);
+
             if (Str::startsWith($path, ['http', 'https', '//'])) {
                 $href = $path;
             } else {
                 $href = asset($path);
             }
 
-            $integrity = SriFacade::html($path, $crossOrigin);
-
-            if (Str::endsWith($path, 'css')) {
-                return "<link href='{$href}' rel='stylesheet' {$integrity}>";
-            } elseif (Str::endsWith($path, 'js')) {
-                return "<script src='{$href}' {$integrity}></script>";
-            } else {
-                throw new \Exception('Invalid file');
-            }
+            return $this->generate($path, $href, $crossOrigin);
         });
     }
+
+    /**
+     * Remove apostrophe and quotation mark.
+     *
+     * @param string  $path
+     * @return string
+     *
+     * @throws \Exception
+     */
+    private function clean_path($path): string
+    {
+        $values = ['\'', '"'];
+
+        return str_replace($values, '', $path);
+    }
+
+    /**
+     * Parse and generate the URL.
+     *
+     * @param string  $path
+     * @param string  $href
+     * @param bool    $crossOrigin
+     * @return void
+     *
+     * @throws \Exception
+     */
+    private function generate(string $path, string $href, bool $crossOrigin)
```

Comment:
Same comments as the previous method, so better naming to avoid an explanatory comment.

`parseAndGenerateUrl()` would be a good name.

I would also remove the whole docblock as we have all the information needed in the method definition.